### PR TITLE
[INFRA] Add Dependency Review step to prevent introducing vulnerable dependencies

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- Add actions/dependency-review-action (https://github.com/actions/dependency-review-action) as new step in the dependency workflow, which fails when future PR is bringing vulnerable dependencies.
- configuring `fail-on-severity` to `moderate` level

image

### _How was this patch tested?_
- [x] Pass the Dependency Check CI job

### CODE REVIEW

File: .github/workflows/dep.yml [hunk 1]
```diff
@@ -57,3 +58,7 @@ jobs:
           -pl kyuubi-ctl,kyuubi-server,kyuubi-assembly -am
       - name: Check dependency list
         run: build/dependency.sh
+      - name: Dependency Review
+        uses: actions/dependency-review-action@v3
+        with:
+          fail-on-severity: low
```
Comment (yaooqinn) [inline .github/workflows/dep.yml:64]: Is `low` too strict? Use moderate?

File: .github/workflows/dep.yml [hunk 2]
```diff
@@ -25,6 +25,7 @@ on:
     paths:
       # dependency check happens only pom changes
       - '**/pom.xml'
+      - '.github/workflows/dep.yml'
```
Comment (yaooqinn) [inline .github/workflows/dep.yml:28]: Is this necessary?

File: .github/workflows/dep.yml (hunk 1 above)
Comment (bowenliang123) [reply .github/workflows/dep.yml:64]: The docs of action does not clarify how it classify the level of severity, so I prefer to keep it in `low` the same as the default value. We could lower this level if necessary in future, as it only affects future PRs.

File: .github/workflows/dep.yml (hunk 2 above)
Comment (bowenliang123) [reply .github/workflows/dep.yml:28]: `dep.yml` is designed to check and verify dependencies although doesn't bring changes in dependencies, I think it's good to leave it in paths. As for this PR, this is add to trigger the Dependency Check job.

File: .github/workflows/dep.yml (hunk 1 above)
Comment (yaooqinn) [reply .github/workflows/dep.yml:64]: The risk for vulnerability is determined by the CVSS score. If you go to check the mvn central or NVD, it's common to see an artifact have some low-level vulnerabilities. I guess it is not practical for us to add such a critical rule that blocks PRs frequently

Comment (bowenliang123) [reply .github/workflows/dep.yml]: OK, changed to `moderate`.

**Review by yaooqinn — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**